backend/lib/models/Post.js

```diff
@@ -69,6 +69,16 @@ const postSchema = new Schema(
       ref: "User",
       type: [ObjectId],
     },
+    views: {
+      default: [],
+      ref: "User",
+      type: [ObjectId],
+    },
+    shares: {
+      default: [],
```

Comment:
I don't know if we'll ever need to see which user shared the post, if the only information required is to display the count of shares instead of specific users shared the post, we may use a number instead?